src/compiler/crystal/tools/typed_def_processor.cr

```diff
@@ -3,6 +3,8 @@
 #
 # It is used for `crystal tool context/expand/implementation`.
 module Crystal::TypedDefProcessor
+  @processed_types = Set(Type).new
```

Comment:
This is now needed because otherwise it would recurse endlessly, I'm not sure if it's because of aliases or what... I'm surprised it worked before, whenever I traverse types in code (`AbstractDefChecker`, `RecursiveStructChecker`, etc.) I keep a registry of all processed types.